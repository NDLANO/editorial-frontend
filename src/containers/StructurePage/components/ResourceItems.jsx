/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { injectT } from 'ndla-i18n';
import { ResourceShape } from '../../../shapes';
import Resource from './Resource';
import { deleteTopicResource } from '../../../modules/taxonomy';
import WarningModal from '../../../components/WarningModal';
import { classes } from './ResourceGroup';

class ResourceItems extends React.PureComponent {
  constructor() {
    super();
    this.state = {};
    this.onDelete = this.onDelete.bind(this);
    this.toggleDelete = this.toggleDelete.bind(this);
  }

  async onDelete(id) {
    try {
      this.setState({ deleteId: '' });
      await deleteTopicResource(id);
      this.props.refreshResources();
    } catch (error) {
      console.log(error);
    }
  }

  toggleDelete(id) {
    this.setState({ deleteId: id });
  }

  render() {
    const { contentType, resources, t } = this.props;
    return (
      <ul {...classes('list')}>
        <li {...classes('item')}>
          {resources.map(resource => (
            <Resource
              key={resource.id}
              contentType={contentType}
              name={resource.name}
              onDelete={() => this.toggleDelete(resource.connectionId)}
            />
          ))}
        </li>
        {this.state.deleteId && (
          <WarningModal
            confirmDelete
            text={t('taxonomy.resource.confirmDelete')}
            onContinue={() => this.onDelete(this.state.deleteId)}
            onCancel={() => this.toggleDelete('')}
          />
        )}
      </ul>
    );
  }
}

ResourceItems.propTypes = {
  contentType: PropTypes.string.isRequired,
  resources: PropTypes.arrayOf(ResourceShape),
  classes: PropTypes.func,
  refreshResources: PropTypes.func,
};

export default injectT(ResourceItems);
