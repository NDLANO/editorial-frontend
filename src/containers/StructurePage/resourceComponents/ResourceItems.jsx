/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { ResourceShape } from '../../../shapes';
import Resource from './Resource';
import { deleteTopicResource, updateTopicResource } from '../../../modules/taxonomy';
import { updateRelevanceId } from '../../../util/taxonomyHelpers';
import handleError from '../../../util/handleError';
import MakeDndList from '../../../components/MakeDndList';
import AlertModal from '../../../components/AlertModal';
import { classes } from './ResourceGroup';
import Spinner from '../../../components/Spinner';
import { StructureShape } from '../../../shapes';

const StyledResourceItems = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`;

const StyledErrorMessage = styled.div`
  text-align: center;
`;

class ResourceItems extends React.PureComponent {
  constructor() {
    super();
    this.state = {};
    this.onDelete = this.onDelete.bind(this);
    this.toggleDelete = this.toggleDelete.bind(this);
    this.onDragEnd = this.onDragEnd.bind(this);
  }

  async onDelete(deleteId) {
    try {
      this.setState({ deleteId: '', error: '' });
      await deleteTopicResource(deleteId);
      this.props.refreshResources();
    } catch (e) {
      handleError(e);
      this.setState({
        error: `${this.props.t('taxonomy.errorMessage')}: ${e.message}`,
      });
    }
  }

  async onDragEnd({ destination, source }) {
    if (!destination) {
      return;
    }
    try {
      const { resources, refreshResources } = this.props;
      const { connectionId, primary, relevanceId, rank: currentRank } = resources[source.index];
      const { rank } = resources[destination.index];
      if (currentRank === rank) {
        return;
      }

      this.setState({ loading: true });
      await updateTopicResource(connectionId, {
        primary,
        rank: currentRank > rank ? rank : rank + 1,
        relevanceId,
      });
      await refreshResources();
      this.setState({ loading: false });
    } catch (e) {
      handleError(e.message);
    }
  }

  toggleDelete(deleteId) {
    this.setState({ deleteId });
  }

  render() {
    const { resources, t, currentSubject, structure, locale } = this.props;

    const { deleteId, error, loading } = this.state;

    if (loading) {
      return <Spinner />;
    }
    return (
      <StyledResourceItems {...classes('list')}>
        <MakeDndList onDragEnd={this.onDragEnd} dragHandle>
          {resources.map(resource => (
            <Resource
              resource={resource}
              key={resource.id}
              id={resource.id}
              currentSubject={currentSubject}
              structure={structure}
              onDelete={this.toggleDelete}
              locale={locale}
              updateRelevanceId={updateRelevanceId}
              {...resource}
            />
          ))}
        </MakeDndList>
        {error && (
          <StyledErrorMessage data-testid="inlineEditErrorMessage" {...classes('errorMessage')}>
            {error}
          </StyledErrorMessage>
        )}
        <AlertModal
          show={!!deleteId}
          text={t('taxonomy.resource.confirmDelete')}
          actions={[
            {
              text: t('form.abort'),
              onClick: () => this.toggleDelete(''),
            },
            {
              text: t('alertModal.delete'),
              onClick: () => this.onDelete(deleteId),
            },
          ]}
          onCancel={() => this.toggleDelete('')}
        />
      </StyledResourceItems>
    );
  }
}

ResourceItems.propTypes = {
  resources: PropTypes.arrayOf(ResourceShape),
  classes: PropTypes.func,
  refreshResources: PropTypes.func.isRequired,
  currentSubject: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
  }),
  currentTopic: PropTypes.shape({}),
  structure: PropTypes.arrayOf(StructureShape),
  locale: PropTypes.string,
};

export default withTranslation()(ResourceItems);
