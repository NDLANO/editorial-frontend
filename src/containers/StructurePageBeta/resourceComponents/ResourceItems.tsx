/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { memo, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useQueryClient } from 'react-query';
import { DropResult } from 'react-beautiful-dnd';
import { partition, sortBy, uniqBy } from 'lodash';
import styled from '@emotion/styled';
import Resource from './Resource';
import handleError from '../../../util/handleError';
import {
  useDeleteResourceForNodeMutation,
  usePutResourceForNodeMutation,
} from '../../../modules/nodes/nodeMutations';
import { RESOURCES_WITH_NODE_CONNECTION } from '../../../queryKeys';
import { ResourceWithNodeConnection } from '../../../modules/nodes/nodeApiTypes';
import AlertModal from '../../../components/AlertModal';
import { classes } from './ResourceGroup';
import MakeDndList from '../../../components/MakeDndList';
import { useTaxonomyVersion } from '../../StructureVersion/TaxonomyVersionProvider';

const StyledResourceItems = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`;

const StyledErrorMessage = styled.div`
  text-align: center;
`;

interface Props {
  resources: ResourceWithNodeConnection[];
  currentNodeId: string;
}

const isError = (error: unknown): error is Error => (error as Error).message !== undefined;

const ResourceItems = ({ resources, currentNodeId }: Props) => {
  const { t, i18n } = useTranslation();
  const [deleteId, setDeleteId] = useState<string>('');
  const { taxonomyVersion } = useTaxonomyVersion();

  const qc = useQueryClient();
  const compKey = [RESOURCES_WITH_NODE_CONNECTION, currentNodeId, { language: i18n.language }];
  const deleteNodeResource = useDeleteResourceForNodeMutation({
    onMutate: async ({ id }) => {
      await qc.cancelQueries(compKey);
      const prevData = qc.getQueryData<ResourceWithNodeConnection[]>(compKey) ?? [];
      const withoutDeleted = prevData.filter(res => res.connectionId !== id);
      qc.setQueryData<ResourceWithNodeConnection[]>(compKey, withoutDeleted);
      return prevData;
    },
  });

  const onUpdateRank = async (id: string, newRank: number) => {
    await qc.cancelQueries(compKey);
    const [toUpdate, other] = partition(resources, t => t.connectionId === id);
    const updatedRes: ResourceWithNodeConnection = { ...toUpdate[0], rank: newRank };
    const prevData = qc.getQueryData<ResourceWithNodeConnection[]>(compKey) ?? [];
    const updated = other.map(t => (t.rank >= updatedRes.rank ? { ...t, rank: t.rank + 1 } : t));
    const newArr = sortBy([...updated, updatedRes], 'rank');
    const allResources = uniqBy<ResourceWithNodeConnection>([...newArr, ...prevData], 'id');
    qc.setQueryData<ResourceWithNodeConnection[]>(compKey, allResources);
    return resources;
  };

  const { mutateAsync: updateNodeResource } = usePutResourceForNodeMutation({
    onMutate: ({ id, body }) => onUpdateRank(id, body.rank as number),
    onError: e => handleError(e),
    onSuccess: () => qc.invalidateQueries(compKey),
  });

  const onDelete = async (deleteId: string) => {
    setDeleteId('');
    await deleteNodeResource.mutateAsync(
      { id: deleteId, taxonomyVersion },
      { onSuccess: () => qc.invalidateQueries(compKey) },
    );
  };

  const onDragEnd = async ({ destination, source }: DropResult) => {
    if (!destination) return;
    const { connectionId, primary, relevanceId, rank: currentRank } = resources[source.index];
    const { rank } = resources[destination.index];
    if (currentRank === rank) {
      return;
    }
    await updateNodeResource({
      id: connectionId,
      body: {
        primary,
        rank: currentRank > rank ? rank : rank + 1,
        relevanceId,
      },
      taxonomyVersion,
    });
  };

  const toggleDelete = (newDeleteId: string) => {
    setDeleteId(newDeleteId);
  };

  return (
    <StyledResourceItems {...classes('list')}>
      <MakeDndList onDragEnd={onDragEnd} dragHandle disableDnd={false}>
        {resources.map(resource => (
          <Resource
            currentNodeId={currentNodeId}
            id={resource.id}
            connectionId={resource.connectionId}
            resource={resource}
            key={resource.id}
            onDelete={toggleDelete}
          />
        ))}
      </MakeDndList>
      {deleteNodeResource.error && isError(deleteNodeResource.error) && (
        <StyledErrorMessage data-testid="inlineEditErrorMessage" {...classes('errorMessage')}>
          {`${t('taxonomy.errorMessage')}: ${deleteNodeResource.error.message}`}
        </StyledErrorMessage>
      )}
      <AlertModal
        show={!!deleteId}
        text={t('taxonomy.resource.confirmDelete')}
        actions={[
          {
            text: t('form.abort'),
            onClick: () => toggleDelete(''),
          },
          {
            text: t('alertModal.delete'),
            onClick: () => onDelete(deleteId!),
          },
        ]}
        onCancel={() => toggleDelete('')}
      />
    </StyledResourceItems>
  );
};

export default memo(ResourceItems);
