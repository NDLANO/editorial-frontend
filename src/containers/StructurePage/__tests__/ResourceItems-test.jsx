/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import nock from 'nock';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import ResourceItems from '../resourceComponents/ResourceItems';
import {
  supplementaryResourcesMock,
  coreResourcesMock,
} from '../../../util/__tests__/taxonomyMocks';
import IntlWrapper from '../../../util/__tests__/IntlWrapper';

afterEach(cleanup);

const qc = new QueryClient();

const wrapper = () =>
  render(
    <MemoryRouter>
      <IntlWrapper>
        <QueryClientProvider client={qc}>
          <ResourceItems
            resources={[...supplementaryResourcesMock, ...coreResourcesMock]}
            contentType="topic-article"
            locale="nb"
            refreshResources={() => {}}
          />
        </QueryClientProvider>
      </IntlWrapper>
    </MemoryRouter>,
  );

test('matches snapshot', () => {
  nock('http://ndla-api')
    .persist()
    .get('/learningpath-api/v2/learningpaths/401?language=nb&fallback=true')
    .reply(200, {});

  nock('http://ndla-api')
    .persist()
    .get('/learningpath-api/v2/learningpaths/508?language=nb&fallback=true')
    .reply(200, {});

  nock('http://ndla-api')
    .persist()
    .get('/learningpath-api/v2/learningpaths/813?language=nb&fallback=true')
    .reply(200, {});

  nock('http://ndla-api')
    .persist()
    .get('/learningpath-api/v2/learningpaths/78?language=nb&fallback=true')
    .reply(200, {});

  nock('http://ndla-api')
    .persist()
    .get('/learningpath-api/v2/learningpaths/89?language=nb&fallback=true')
    .reply(200, {});

  const { container } = wrapper();
  expect(container.firstChild).toMatchSnapshot();
});
