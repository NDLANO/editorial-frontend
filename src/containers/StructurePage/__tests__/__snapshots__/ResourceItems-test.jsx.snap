// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.emotion-0 {
  color: #20588f;
}

.emotion-2 {
  display: inline-block;
  color: #ffffff;
  background-color: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  padding: 4px 13px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 700;
  -webkit-transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  box-shadow: none;
  text-align: center;
  -webkit-transition: background-color none;
  transition: background-color none;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-2:hover,
.emotion-2:focus {
  color: white;
  background-color: #184673;
  border: 2px solid #184673;
}

.emotion-2[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-2:focus {
  box-shadow: 0 0 2px #20588f;
}

.emotion-2:hover,
.emotion-2:active,
.emotion-2:disabled,
.emotion-2:focus {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-2:focus {
  outline-width: medium;
}

<ul
  class="c-topic-resource__list"
>
  <div
    class="c-dnd-list__drop-zone"
  >
    <div
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-draggable="0"
    >
      <div
        class="c-topic-resource__text o-flag o-flag--top"
        data-testid="resource-type-topic-article"
      >
        <div
          aria-roledescription="Draggable item. Press space bar to lift"
          class="c-topic-resource__icon o-flag__img"
          data-react-beautiful-dnd-drag-handle="0"
          draggable="false"
          tabindex="0"
        >
          <div
            class="c-content-type-badge c-content-type-badge--topic c-content-type-badge--small c-content-type-badge--background c-content-type-badge--border"
          >
            <svg
              class="c-icon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                Subject
              </title>
              <g>
                <title>
                  Emne
                </title>
                <defs />
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class="c-topic-resource__body o-flag__body"
        >
          <h1
            class="c-topic-resource__title emotion-0 emotion-1"
          >
            Kontrast og balanse i grafisk design
          </h1>
        </div>
        <button
          class="emotion-2 emotion-3"
          type="button"
        >
          <svg
            class="c-icon c-topic-resource__deleteIcon"
            data-license="Apache License 2.0"
            data-source="Material Design"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 24 24"
            width="1em"
          >
            <title>
              RemoveCircle
            </title>
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-draggable="0"
    >
      <div
        class="c-topic-resource__text o-flag o-flag--top"
        data-testid="resource-type-topic-article"
      >
        <div
          aria-roledescription="Draggable item. Press space bar to lift"
          class="c-topic-resource__icon o-flag__img"
          data-react-beautiful-dnd-drag-handle="0"
          draggable="false"
          tabindex="0"
        >
          <div
            class="c-content-type-badge c-content-type-badge--topic c-content-type-badge--small c-content-type-badge--background c-content-type-badge--border"
          >
            <svg
              class="c-icon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                Subject
              </title>
              <g>
                <title>
                  Emne
                </title>
                <defs />
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class="c-topic-resource__body o-flag__body"
        >
          <a
            href="/subject-matter/topic-article/89/edit"
          >
            <h1
              class="c-topic-resource__title emotion-0 emotion-1"
            >
              Hvorfor bruker vi mediene?
            </h1>
          </a>
        </div>
        <button
          class="emotion-2 emotion-3"
          type="button"
        >
          <svg
            class="c-icon c-topic-resource__deleteIcon"
            data-license="Apache License 2.0"
            data-source="Material Design"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 24 24"
            width="1em"
          >
            <title>
              RemoveCircle
            </title>
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-draggable="0"
    >
      <div
        class="c-topic-resource__text o-flag o-flag--top"
        data-testid="resource-type-topic-article"
      >
        <div
          aria-roledescription="Draggable item. Press space bar to lift"
          class="c-topic-resource__icon o-flag__img"
          data-react-beautiful-dnd-drag-handle="0"
          draggable="false"
          tabindex="0"
        >
          <div
            class="c-content-type-badge c-content-type-badge--topic c-content-type-badge--small c-content-type-badge--background c-content-type-badge--border"
          >
            <svg
              class="c-icon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                Subject
              </title>
              <g>
                <title>
                  Emne
                </title>
                <defs />
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class="c-topic-resource__body o-flag__body"
        >
          <a
            href="/subject-matter/topic-article/78/edit"
          >
            <h1
              class="c-topic-resource__title emotion-0 emotion-1"
            >
              Medieundersøkelser og mediebruk
            </h1>
          </a>
        </div>
        <button
          class="emotion-2 emotion-3"
          type="button"
        >
          <svg
            class="c-icon c-topic-resource__deleteIcon"
            data-license="Apache License 2.0"
            data-source="Material Design"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 24 24"
            width="1em"
          >
            <title>
              RemoveCircle
            </title>
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-draggable="0"
    >
      <div
        class="c-topic-resource__text o-flag o-flag--top"
        data-testid="resource-type-topic-article"
      >
        <div
          aria-roledescription="Draggable item. Press space bar to lift"
          class="c-topic-resource__icon o-flag__img"
          data-react-beautiful-dnd-drag-handle="0"
          draggable="false"
          tabindex="0"
        >
          <div
            class="c-content-type-badge c-content-type-badge--topic c-content-type-badge--small c-content-type-badge--background c-content-type-badge--border"
          >
            <svg
              class="c-icon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                Subject
              </title>
              <g>
                <title>
                  Emne
                </title>
                <defs />
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class="c-topic-resource__body o-flag__body"
        >
          <a
            href="/subject-matter/topic-article/813/edit"
          >
            <h1
              class="c-topic-resource__title emotion-0 emotion-1"
            >
              Mediebruk og statistikk
            </h1>
          </a>
        </div>
        <button
          class="emotion-2 emotion-3"
          type="button"
        >
          <svg
            class="c-icon c-topic-resource__deleteIcon"
            data-license="Apache License 2.0"
            data-source="Material Design"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 24 24"
            width="1em"
          >
            <title>
              RemoveCircle
            </title>
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-draggable="0"
    >
      <div
        class="c-topic-resource__text o-flag o-flag--top"
        data-testid="resource-type-topic-article"
      >
        <div
          aria-roledescription="Draggable item. Press space bar to lift"
          class="c-topic-resource__icon o-flag__img"
          data-react-beautiful-dnd-drag-handle="0"
          draggable="false"
          tabindex="0"
        >
          <div
            class="c-content-type-badge c-content-type-badge--topic c-content-type-badge--small c-content-type-badge--background c-content-type-badge--border"
          >
            <svg
              class="c-icon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                Subject
              </title>
              <g>
                <title>
                  Emne
                </title>
                <defs />
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class="c-topic-resource__body o-flag__body"
        >
          <a
            href="/subject-matter/topic-article/508/edit"
          >
            <h1
              class="c-topic-resource__title emotion-0 emotion-1"
            >
              Identitet og mediebruk
            </h1>
          </a>
        </div>
        <button
          class="emotion-2 emotion-3"
          type="button"
        >
          <svg
            class="c-icon c-topic-resource__deleteIcon"
            data-license="Apache License 2.0"
            data-source="Material Design"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 24 24"
            width="1em"
          >
            <title>
              RemoveCircle
            </title>
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-draggable="0"
    >
      <div
        class="c-topic-resource__text o-flag o-flag--top"
        data-testid="resource-type-topic-article"
      >
        <div
          aria-roledescription="Draggable item. Press space bar to lift"
          class="c-topic-resource__icon o-flag__img"
          data-react-beautiful-dnd-drag-handle="0"
          draggable="false"
          tabindex="0"
        >
          <div
            class="c-content-type-badge c-content-type-badge--topic c-content-type-badge--small c-content-type-badge--background c-content-type-badge--border"
          >
            <svg
              class="c-icon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                Subject
              </title>
              <g>
                <title>
                  Emne
                </title>
                <defs />
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class="c-topic-resource__body o-flag__body"
        >
          <a
            href="/subject-matter/topic-article/401/edit"
          >
            <h1
              class="c-topic-resource__title emotion-0 emotion-1"
            >
              Mediepåvirkning
            </h1>
          </a>
        </div>
        <button
          class="emotion-2 emotion-3"
          type="button"
        >
          <svg
            class="c-icon c-topic-resource__deleteIcon"
            data-license="Apache License 2.0"
            data-source="Material Design"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 24 24"
            width="1em"
          >
            <title>
              RemoveCircle
            </title>
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
</ul>
`;
